src/database/schema/ChatSchema.ts

```diff
@@ -424,7 +428,9 @@ class ChatSchema implements BaseDbSchema {
       console.log(e);
     }
     try {
-      attachmentJson = JSON.stringify(data?.message?.attachments);
+      if (isSavingAttachmentAllowed(data?.message?.message_type)) {
+        attachmentJson = JSON.stringify(data?.message?.attachments);
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The condition for saving attachments has been refactored into a separate function `isSavingAttachmentAllowed`. This change also adds an additional check before stringifying the attachments, which can prevent potential errors when `data?.message?.attachments` is undefined or not valid JSON.
```diff
- attachmentJson = JSON.stringify(data?.message?.attachments);
+ if (isSavingAttachmentAllowed(data?.message?.message_type)) {
+   attachmentJson = JSON.stringify(data?.message?.attachments);
+ }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->